### PR TITLE
Fix isPragmaComponentWrapper

### LINKED ISSUE #2124: Cannot read property 'name' of undefined in isPragmaComponentWrapper
Error existed in versions 7.12.0-7.12.1.
Version 7.12.2 was ok.
It's present again in version 7.12.3.

> TypeError: Cannot read property 'name' of undefined
>     at Object.isPragmaComponentWrapper (<project_root>/node_modules/eslint-plugin-react/lib/util/Components.js:405:66)

### PR DESCRIPTION
**Pull request by CrOrc**
In isPragmaComponentWrapper check name of callee object if awailable.

Fixes #2124

### CODE REVIEW

File: lib/util/Components.js [hunk 1]
```diff
@@ -402,7 +402,7 @@ function componentRule(rule, context) {
       const propertyNames = ['forwardRef', 'memo'];
       const calleeObject = node.callee.object;
       if (calleeObject) {
-        return arrayIncludes(propertyNames, node.callee.property.name) && node.callee.object.name === pragma;
```
Comment (ljharb) [inline lib/util/Components.js]: This isn’t equivalent, because the first one has .property and not .object.

Comment (CrOrc) [reply lib/util/Components.js]: Maybe I should add check for both node.callee.property and node.callee.object?

Comment (ljharb) [reply lib/util/Components.js]: I’m not really sure - a regression test would help make that clear.

Comment (CrOrc) [reply lib/util/Components.js]: Added a regression test

**Comment by rpellerin** [comment]
When will this be released?

**Comment by ljharb** [comment]
[v7.12.4](https://github.com/yannickcr/eslint-plugin-react/releases/tag/v7.12.4) has been released.

**Pull request merged**